src/context/WalletProvider.tsx

```diff
@@ -416,6 +424,66 @@ const WalletProvider = function WalletProvider({ children }: { children: React.R
     return false
   }
 
+  const connectSafe = async (): Promise<boolean> => {
+    updateWalletState('Safe', { status: WalletStatusEnum.LOADING })
+
+    try {
+      const { sdk, safe } = useSafeAppsSDK()
```

Comment:
hey!

https://reactjs.org/docs/hooks-rules.html#only-call-hooks-at-the-top-level:

```
Don’t call Hooks inside loops, conditions, or nested functions. Instead, always use Hooks at the top level of your React function, before any early returns. By following this rule, you ensure that Hooks are called in the same order each time a component renders.
```